x/staking/keeper/keeper.go

```diff
@@ -172,3 +179,160 @@ func (k Keeper) GetValidatorUpdates(ctx context.Context) ([]abci.ValidatorUpdate
 
 	return valUpdates.Updates, nil
 }
+
+// Copied from baseapp/abci_utils.go
+func (k Keeper) ValidateVoteExtensions(ctx sdk.Context,
+	currentHeight int64,
+	chainID string,
+	extCommit abci.ExtendedCommitInfo) error {
+	cp := ctx.ConsensusParams()
+	extsEnabled := cp.Abci != nil && currentHeight >= cp.Abci.VoteExtensionsEnableHeight && cp.Abci.VoteExtensionsEnableHeight != 0
+
+	marshalDelimitedFn := func(msg proto.Message) ([]byte, error) {
+		var buf bytes.Buffer
+		if err := protoio.NewDelimitedWriter(&buf).WriteMsg(msg); err != nil {
+			return nil, err
+		}
+
+		return buf.Bytes(), nil
+	}
+
+	sumVP := math.NewInt(0)
+	for _, vote := range extCommit.Votes {
+		if !extsEnabled {
+			if len(vote.VoteExtension) > 0 {
+				return fmt.Errorf("vote extensions disabled; received non-empty vote extension at height %d", currentHeight)
+			}
+			if len(vote.ExtensionSignature) > 0 {
+				return fmt.Errorf("vote extensions disabled; received non-empty vote extension signature at height %d", currentHeight)
+			}
+
+			continue
+		}
+
+		if len(vote.ExtensionSignature) == 0 {
+			return fmt.Errorf("vote extensions enabled; received empty vote extension signature at height %d", currentHeight)
```

Comment:
Does it need to be error? Can we have no vote extensions for some txs (not side-txs)?

Reply:
Yes we can , but they have to be signed. cc: @lasarojc